sqlmesh/core/console.py

```diff
@@ -1550,10 +1569,19 @@ def show_model_difference_summary(
                         self._print(f"```diff\n{context_diff.text_diff(snapshot.name)}\n```")
             if indirectly_modified:
                 self._print("\n**Indirectly Modified:**")
-                for snapshot in sorted(indirectly_modified):
+                modified_length = len(indirectly_modified)
+                if (
+                    not self.verbose
+                    and modified_length > self.INDIRECTLY_MODIFIED_DISPLAY_THRESHOLD
+                ):
                     self._print(
-                        f"- `{snapshot.display_name(environment_naming_info, default_catalog, dialect=self.dialect)}`"
+                        f"- `{indirectly_modified[0].display_name(environment_naming_info, default_catalog, dialect=self.dialect)}`\n- `.... {modified_length-2} more ....`\n- `{indirectly_modified[-1].display_name(environment_naming_info, default_catalog, dialect=self.dialect)}`"
```

Comment:
Let's make this a bit easier to read. Another concern is that the order here won't necessarily match that in the `else` branch, because we don't sort here. I'd sort beforehand.

Reply:
Thank you, refactored the code to address these